### PR TITLE
DM report/emergency confirmation.

### LINKED ISSUE #90: [FEAT] DM report confirmation
# Overview
## Describe the solution you'd like
DM the "reportee" to confirm that their report has been received.

## Is this suggestion based on an existing problem? If so, please describe it
Users cannot be certain that their report has been received.

### PR DESCRIPTION
**Pull request by portalBlock**
# Initial Checklist
- [x] Has @tigattack / @kdavis / @issy been added as a reviewer?
- [x] If applicable, have the relevant project(s), milestone(s), and label(s) been applied?
- [ ] If applicable, have you added details of the cog to the readme as per [README.md](https://github.com/rHomelab/LabBot-Cogs/blob/develop/README.md#cog-summaries)?

# Details
* **Does this resolve an issue?**
Resolves #90 

### New Features
* Adds feature to notify users via DM when reports (including emergencies) are filed.

### Breaking Changes
* No known breaking changes.

### CODE REVIEW

**Comment by tigattack** [comment]
Could there be a switch for this functionality?

Example: `[p]report confirm true|false`

File: report/report.py [hunk 1]
```diff
@@ -87,6 +93,12 @@ async def cmd_emergency(self, ctx: commands.Context, *, message: str = None):
             mod_pings = " ".join([i.mention for i in log.members if not i.bot])
         await log.send(content=mod_pings, embed=data)
 
+        report_reply = self.make_reporter_reply(ctx, message, True)
+        if ctx.channel.type == discord.ChannelType.private:
```
Comment (issy) [inline report/report.py:97]: This is a guild only command, so the channel type will never be private

File: report/report.py [hunk 2]
```diff
@@ -87,6 +93,12 @@ async def cmd_emergency(self, ctx: commands.Context, *, message: str = None):
...
+            await ctx.send(embed=report_reply)
+        else:
+            await ctx.author.send(embed=report_reply)
```
Comment (issy) [inline report/report.py:100]: This needs some error handling. If a user doesn't allow DMs it should fail silently without spitting an error message
(redbot has a default error handler that'll throw a message in ctx.channel like `there was an error in the report command`  if errors aren't caught.

File: report/report.py [hunk 3]
```diff
@@ -100,3 +112,14 @@ def make_report_embed(self, ctx: commands.Context, message: str):
             name="Message", value=escape(message or "<no message>"), inline=False
         )
         return data
+
+    def make_reporter_reply(self, ctx: commands.Context, message: str, emergency: bool) -> discord.Embed:
+        report_reply_embed = discord.Embed(
+            title="Report Received", colour=discord.Color.orange() if emergency else ctx.guild.me.colour
+        )
+        report_reply_embed.add_field(name="Server", value=ctx.guild.name)
```
Comment (issy) [inline report/report.py:120]: It'd be nice to have these fields be inline, to fit the embed that is sent to modchat

**Review by issy — CHANGES_REQUESTED** [review]

File: report/report.py (hunk 2 above)
Comment (tigattack) [reply report/report.py:100]: May be a bit OTT but what about creating a temporary channel, readable only by the reporter and mods+, where a confirmation message is sent, and the reporter is mentioned? Could add a react for confirmation, triggering nukage of channel once confirmed, and a timeout of, say, 6 hours, after which the channel is automatically nuked.

Comment (issy) [reply report/report.py:100]: Personally not a fan of that from a UX POV. It seems like it would be annoying and OTT, personally

Comment (portalBlock) [reply report/report.py:98]: @issy the error handling has been added.

@tigattack I agree with Issy. I think it's easy enough to follow up via DM if we need to carry on the conversation.

File: report/report.py (hunk 1 above)
Comment (portalBlock) [reply report/report.py]: DM logic has been removed.

File: report/report.py (hunk 3 above)
Comment (portalBlock) [reply report/report.py]: Made the embed nearly identical to the mod alert.

File: report/report.py (hunk 2 above)
Comment (tigattack) [reply report/report.py:98]: Sounds good 👍

**Review by issy — APPROVED** [review]

**Review by issy — APPROVED** [review]

**Pull request merged**